src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleAssignments.java

```diff
@@ -14,21 +18,44 @@
 @Builder
 @Data
 @AllArgsConstructor
-@NoArgsConstructor
 public class RoleAssignments {
     private List<RoleAssignment> roleAssignments;
 
+    private static final String ORGANISATION = "Organisation";
+    private final CaseDefinitionRepository caseDefinitionRepository;
+
+    @Autowired
+    public RoleAssignments(@Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                               CaseDefinitionRepository caseDefinitionRepository) {
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
     @JsonIgnore
     public List<String> getJurisdictions() {
-        return roleAssignments.stream()
-            .map(RoleAssignment::getAttributes)
-            .filter(Objects::nonNull)
-            .map(RoleAssignmentAttributes::getJurisdiction)
-            .filter(Objects::nonNull)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .distinct()
-            .collect(Collectors.toList());
+        List<String> jurisdictions = new ArrayList<>();
+
+        roleAssignments.stream()
+            .filter(roleAssignment -> !roleAssignment.getGrantType().equals(GrantType.BASIC.name()))
+            .collect(Collectors.toList())
+            .forEach(roleAssignmentAttributes -> {
+                if (roleAssignmentAttributes.getAttributes() != null) {
+                    if (Objects.nonNull(roleAssignmentAttributes.getAttributes().getJurisdiction())) {
+                        String jurisdiction = roleAssignmentAttributes
+                            .getAttributes().getJurisdiction().orElse(null);
+                        jurisdictions.add(jurisdiction);
+                    } else if (Objects.nonNull(roleAssignmentAttributes.getAttributes().getCaseType())) {
+                        jurisdictions.add(caseDefinitionRepository
+                            .getCaseType(roleAssignmentAttributes
+                                .getAttributes().getCaseType().get()).getJurisdictionId());
+                    } else if (Objects.nonNull(roleAssignmentAttributes.getRoleType())
+                        && roleAssignmentAttributes.getRoleType().contentEquals(ORGANISATION)) {
+                        caseDefinitionRepository.getAllJurisdictionsFromDefinitionStore(Optional.empty())
+                            .forEach(jurisdictionDefinition -> jurisdictions.add(jurisdictionDefinition.getId()));
```

Comment:
If we end up in this `else if`, after adding all jurisdictions we could stop processing the loop as we're not going to get anything else

Reply:
done